hierarchyAPI/get_root_hierarchy_test.go

```diff
@@ -0,0 +1,94 @@
+package hierarchyAPI
+
+import (
+	"testing"
+
+	"fmt"
+	"github.com/ONSdigital/dp-api-tests/testDataSetup/neo4j"
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/gavv/httpexpect"
+	uuid "github.com/satori/go.uuid"
+	. "github.com/smartystreets/goconvey/convey"
+	"net/http"
+	"os"
+)
+
+func TestSuccessfullyGetRootHierarchy(t *testing.T) {
+	instanceID := uuid.NewV4().String()
+	hierarchyAPI := httpexpect.New(t, cfg.HierarchyAPIURL)
+
+	Convey("Given an existing hierarchy", t, func() {
+
+		datastore := neo4j.NewDatastore(cfg.Neo4jAddr, instanceID, neo4j.HierarchyTestData)
+		Convey("When a root hierarchy node is requested", func() {
+
+			err := datastore.Setup()
+			if err != nil {
+				log.ErrorC("Unable to setup test data", err, nil)
+				os.Exit(1)
+			}
+
+			Convey("Then a root hierarchy node is return as a response", func() {
+				response := hierarchyAPI.GET("/hierarchies/{instance_id}/{dimension_name}", instanceID, "aggregate").
+					Expect().Status(http.StatusOK).JSON().Object()
+
+				// Check root node
+				response.Value("has_data").Boolean().Equal(true)
+				response.Value("no_of_children").Number().Equal(12)
+				selfLink := response.Value("links").Object().Value("self").Object()
+				codeLink := response.Value("links").Object().Value("code").Object()
+				codeLink.Value("href").String().
+					Equal("http://localhost:22400/code-list/e44de4c4-d39e-4e2f-942b-3ca10584d078/code/cpi1dim1A0")
+				selfLink.Value("href").String().
+					Equal(fmt.Sprintf("http://localhost:22600/hierarchies/%s/aggregate", instanceID))
+
+				// Check first child node
+				first := response.Value("children").Array().First().Object()
+				first.Value("has_data").Boolean().True()
+				first.Value("label").String().Equal("01 Food and non-alcoholic beverages")
+				first.Value("no_of_children").Number().Equal(2)
+				firstSelfLink := first.Value("links").Object().Value("self").Object()
+				firstCodeLink := first.Value("links").Object().Value("code").Object()
+				firstCodeLink.Value("href").String().
+					Equal("http://localhost:22400/code-list/e44de4c4-d39e-4e2f-942b-3ca10584d078/code/cpi1dim1T10000")
+				firstSelfLink.Value("href").String().
+					Equal(fmt.Sprintf("http://localhost:22600/hierarchies/%s/aggregate/cpi1dim1T10000", instanceID))
+			})
+		})
+
+		err := datastore.TeardownHierarchy()
+		if err != nil {
+			log.ErrorC("Unable to tear down test data", err, nil)
+			os.Exit(1)
+		}
+	})
+}
+
+func TestErrorStatesGetRootHierarchy(t *testing.T) {
```

Comment:
Again a test missing to check error is returned when the instance_id exists but the dimension name is incorrect.

And once again i think the spec is wrong, if instance doesn't exist then return status 400 bad request, if dimension name not found then return status 404 not found.